### PR TITLE
Update docker daemon to dockerd

### PR DESCRIPTION
**Pull request by yuexiao-wang**
**- What I did**
1. Update 'docker daemon' to 'dockerd'
2. Remove 'sudo'  about Windows container in dockerd.md

Signed-off-by: yuexiao-wang wang.yuexiao@zte.com.cn

### CODE REVIEW

File: docs/reference/commandline/dockerd.md [hunk 1]
```diff
@@ -731,7 +731,7 @@ Also Windows Container makes use of `--exec-opt` for special purpose. Docker use
 can specify default container isolation technology with this, for example:
 
 ```bash
-$ sudo dockerd --exec-opt isolation=hyperv
+$ dockerd --exec-opt isolation=hyperv
```
Comment (AkihiroSuda) [inline docs/reference/commandline/dockerd.md]: I wonder this should be

  \```powershell
  \> dockerd --exec-opt isolation=hyperv
  \```

(If so, other Windows-specific codes should be updated as well)

cc @jhowardmsft

Comment (yuexiao-wang) [inline docs/reference/commandline/dockerd.md]: I agree that.  

If need to update , I am willing to open a new PR.

**Comment by jhowardmsft** [comment]
Strictly, any shell that can launch an executable is going to work on Windows. Including bash, PowerShell, or cmd. But agree it would be better to use a native shell such as PowerShell, rather than bash, for Windows specific examples.

**Comment by yuexiao-wang** [comment]
@AkihiroSuda @jhowardmsft PTAL
It has been done  for this PR, but windowsRS1 is failed.
I open a new PR #985 to  use a powershell  instead of bash shell in docker.github.io for Windows specific examples

**Comment by LK4D4** [comment]
ping @thaJeztah

**Comment by yuexiao-wang** [comment]
windowRSi is failed as below:
01:07:42 re-exec error: exit status 1: output: BackupWrite \\?\D:\CI\CI-3aada7fb0\daemon\windowsfilter\17afa1934dc262f4e8bebead62697824301b42c3793618f9106d174e18523b3b\Files\ProgramData\Microsoft\Windows Defender\Definition Updates\Default\MpAvBase.vdm: There is not enough space on the disk.

File: hack/Jenkins/W2L/setup.sh [hunk 2]
```diff
@@ -188,7 +188,7 @@ if [ $ec -eq 0 ]; then
 FROM docker:$COMMITHASH
 RUN hack/make.sh binary
 RUN cp bundles/latest/binary/docker /bin/docker 
-CMD docker daemon -D -H tcp://0.0.0.0:$port_inner $daemon_extra_args
+CMD dockerd -D -H tcp://0.0.0.0:$port_inner $daemon_extra_args
```
Comment (thaJeztah) [inline hack/Jenkins/W2L/setup.sh]: @johnstep could you check this one?

**Review by thaJeztah — APPROVED** [review]
changes LGTM, one question for @johnstep

**Comment by yuexiao-wang** [comment]
ping @johnstep
Is there any suggestion?

**Comment by yuexiao-wang** [comment]
ping @johnstep

File: hack/Jenkins/W2L/setup.sh (hunk 2 above)
Comment (jhowardmsft) [inline hack/Jenkins/W2L/setup.sh]: Looks fine.

**Comment by jhowardmsft** [comment]
LGTM. (I've rebuilt the RS1 machines this morning and queued up a re-run.)

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Pull request merged**